### ISSUE TITLE
Improve Link type for better inference

### ISSUE THREAD

**Issue by medihack** [tatethurston/nextjs-routes#111]
I am using Next.js with nextjs-routes together with Mantine. Mantine has a component `Menu.Item` where all its props [are inferred by a `component` prop](https://mantine.dev/core/menu/#custom-component-as-menuitem). Unfortunately, this fails when setting `component={Link}` as it always infers that the `href` of `Link` is only of type `Query` (and doesn't consider `Route` and `StaticRoute`). 

Currently, `Link` is defined this way:
```ts
  declare function Link(
    props: PropsWithChildren<LinkProps<Route>>
  ): LinkReactElement;
  declare function Link(
    props: PropsWithChildren<LinkProps<StaticRoute>>
  ): LinkReactElement;
  declare function Link(
    props: PropsWithChildren<LinkProps<Query>>
  ): LinkReactElement;

  export default Link;
```

But the issue would be fixed if it would be defined this way:
```ts
  declare function Link(
    props: PropsWithChildren<LinkProps<Route | StaticRoute | Query>>
  ): LinkReactElement;

  export default Link;
```

I can't think of any drawbacks, but I am not a TS expert.

**Comment by tatethurston** [comment]
@medihack This should already be fixed on version 1.0.4. Could you confirm the version you’re using?

**Comment by medihack** [comment]
I am using 1.0.4. Just installed it yesterday for the first time. I also tried to delete it, but it is re-generated in the same way.

From my pnpm-lock.yaml:
```
  /nextjs-routes/1.0.4_next@13.1.1:
    resolution: {integrity: sha512-BdlLv1sdiH2+qXWsWrrVfW8N1dR3iV0pjUvpO4JcQyt51fLUwVPX+G5uABA/tLzkWhK3MPUcXUoc3fILfi2cmg==}
    peerDependencies:
      next: '*'
    dependencies:
      chokidar: 3.5.3
      next: 13.1.1_7xlrwlvvs7cv2obrs6a5y6oxxq
    dev: false
```

**Comment by evgenyboxer** [comment]
Facing the same issue.

**Comment by tatethurston** [comment]
Thanks @medihack and @evgenyboxer, I published v1.0.5 which removes the function overloading definition for Link and should resolve this.
